lib/bandit/websocket/connection.ex

```diff
@@ -53,7 +29,7 @@ defmodule Bandit.WebSocket.Connection do
 
       %Frame.Text{fin: true} = frame ->
         if String.valid?(frame.data) do
-          connection.sock.handle_text_frame(frame.data, socket, connection.sock_state)
+          connection.sock.handle_in({frame.data, opcode: :text}, connection.sock_state)
```

Comment:
More of a `Sock` issue, but what is the intention of using a keyword list as second tuple argument and not just the opcode?

Reply:
It's a concession to parity with the existing `Phoenix.Socket.Transport` behaviour (specifically, https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/socket/transport.ex#L163-L200). I agree that it's a bit awkward (especially as it doesn't have parity with the shape of return values in `{:push, ...` and `{:reply, ...` tuples), but it allows Phoenix to serve `Phoenix.Socket` instances as valid, conforming Sock instances.